src/plugins/intel_gpu/src/graph/gemm.cpp

```diff
@@ -189,7 +204,9 @@ std::vector<layout> gemm_inst::transform_input_layouts(const std::shared_ptr<con
 
     std::vector<layout> layouts = input_layouts;
     layouts[0].set_partial_shape(transposed_input0_pshape);
+    layouts[0].data_padding = get_transposed_padding(layouts[0].data_padding, input_rank, real_in_rank);
     layouts[1].set_partial_shape(transposed_input1_pshape);
+    layouts[1].data_padding = get_transposed_padding(layouts[1].data_padding, input_rank, real_in_rank);
```

Comment:
I think second layout should use its own parameters instead of `input_rank` and `real_in_rank`